src/lib/storage/index/partial_hash/partial_hash_index_impl.cpp

```diff
@@ -0,0 +1,210 @@
+
+const RowID& TableIndexVectorIterator::operator*() const {
+  return *_map_iterator;
+}
+
+TableIndexVectorIterator& TableIndexVectorIterator::operator++() {
+  _map_iterator++;
+  return *this;
+}
+
+bool TableIndexVectorIterator::operator==(const BaseTableIndexIterator& other) const {
+  const auto* const obj = dynamic_cast<const TableIndexVectorIterator*>(&other);
+  return obj && _map_iterator == obj->_map_iterator;
+}
+
+bool TableIndexVectorIterator::operator!=(const BaseTableIndexIterator& other) const {
+  const auto* const obj = dynamic_cast<const TableIndexVectorIterator*>(&other);
+  return !obj || _map_iterator != obj->_map_iterator;
+}
+
+std::shared_ptr<BaseTableIndexIterator> TableIndexVectorIterator::clone() const {
+  return std::make_shared<TableIndexVectorIterator>(*this);
+}
+
+size_t BasePartialHashIndexImpl::insert_entries(
+    const std::vector<std::pair<ChunkID, std::shared_ptr<Chunk>>>& /*chunk*/, const ColumnID /*column_id*/) {
+  return 0;
+}
+
+size_t BasePartialHashIndexImpl::remove_entries(const std::vector<ChunkID>& /*chunk_ids*/) {
+  return 0;
+}
+
+BasePartialHashIndexImpl::Iterator BasePartialHashIndexImpl::cbegin() const {
+  return Iterator(std::make_shared<BaseTableIndexIterator>());
+}
+
+BasePartialHashIndexImpl::Iterator BasePartialHashIndexImpl::cend() const {
+  return Iterator(std::make_shared<BaseTableIndexIterator>());
+}
+
+BasePartialHashIndexImpl::Iterator BasePartialHashIndexImpl::null_cbegin() const {
+  return Iterator(std::make_shared<BaseTableIndexIterator>());
+}
+
+BasePartialHashIndexImpl::Iterator BasePartialHashIndexImpl::null_cend() const {
+  return Iterator(std::make_shared<BaseTableIndexIterator>());
+}
+
+BasePartialHashIndexImpl::IteratorPair BasePartialHashIndexImpl::range_equals(const AllTypeVariant& value) const {
+  return std::make_pair(Iterator(std::make_shared<BaseTableIndexIterator>()),
+                        Iterator(std::make_shared<BaseTableIndexIterator>()));
+}
+
+std::pair<BasePartialHashIndexImpl::IteratorPair, BasePartialHashIndexImpl::IteratorPair>
+BasePartialHashIndexImpl::range_not_equals(const AllTypeVariant& value) const {
+  return std::make_pair(range_equals(value), range_equals(value));
+}
+
+bool BasePartialHashIndexImpl::is_index_for(const ColumnID column_id) const {
+  return false;
+}
+
+std::unordered_set<ChunkID> BasePartialHashIndexImpl::get_indexed_chunk_ids() const {
+  return std::unordered_set<ChunkID>{};
+}
+
+template <typename DataType>
+PartialHashIndexImpl<DataType>::PartialHashIndexImpl(
+    const std::vector<std::pair<ChunkID, std::shared_ptr<Chunk>>>& chunks_to_index, const ColumnID column_id)
+    : BasePartialHashIndexImpl() {
+  insert_entries(chunks_to_index, column_id);
+}
+
+template <typename DataType>
+size_t PartialHashIndexImpl<DataType>::insert_entries(
+    const std::vector<std::pair<ChunkID, std::shared_ptr<Chunk>>>& chunks_to_index, const ColumnID column_id) {
+  const size_t size_before = _indexed_chunk_ids.size();
```

Comment:
I would assume `size` to tell me the number of indexes rows, not chunks. `chunk_count_before`?

Reply:
auto